### ISSUE TITLE
Heap corruption in storage

### ISSUE THREAD

**Issue by andrew-lee-work** [dfinity/cdk-rs#73]
`storage` can lead to memory corruption when a dangling reference is created.

https://github.com/dfinity/cdk-rs/blob/5807d2f7b523f630eddd69acd4b245a8b129eff9/src/ic-cdk/src/storage.rs#L4-L19

Example from @roman-kashitsyn:
```
fn f() {
    let v = storage::get::<Vec<String>>();
    let v_0_ref = &v[0];
    g();
    h(v_0_ref); // dangling reference: g invalidated it
}
fn g() {
    let v = storage::get_mut::<Vec<String>>();
    v.push("hello".to_string());
    v.push("world".to_string());
}
```

**Comment by hansl** [comment]
Isn't this a duplicate of #71?

**Comment by andrew-lee-work** [comment]
#71 is only about documenting the unsafe behavior, in case you decide not to change the behavior or if it still has unsafe block after fixing.

**Comment by adamspofford-dfinity** [comment]
Closed in #215.
